### ISSUE TITLE
bug: hummock event handler panicked at `should find`

### ISSUE THREAD

**Issue by BugenZhao** [risingwavelabs/risingwave#6200]
During the Nexmark longevity [test](https://buildkite.com/risingwave-test/longevity-test/builds/144#01843f38-fd5c-42fb-8834-d704924ea153)...

```
{
    "log": "thread 'risingwave-main' panicked at 'should find', src/storage/src/hummock/event_handler/hummock_event_handler.rs:184:25\n",
    "stream": "stderr",
    "kubernetes": {
        "pod_name": "risingwave-compute-0",
        "namespace_name": "rwc-3-longevity-20221103-204020",
        "pod_id": "330cec19-2d33-489d-8f75-0025475e8f55",
        "host": "ip-10-0-5-104.ap-southeast-1.compute.internal",
        "container_name": "compute",
        "docker_id": "8cce0cb2e237288c9df034d1609211d57f43ec18b08e45da1890deab316a9dfe",
        "container_hash": "ghcr.io/risingwavelabs/risingwave@sha256:cf13817df6e4110e3d314fa843d4be2a89e13302e098ee65670041551df0d4ae",
        "container_image": "ghcr.io/risingwavelabs/risingwave:latest"
    }
}
```

`RUST_BACKTRACE` is not set so there's no backtrace. :(

Full logs [here](https://ap-southeast-1.console.aws.amazon.com/cloudwatch/home?region=ap-southeast-1#logsV2:log-groups/log-group/$252Faws$252Fcontainerinsights$252Frls-apse1-eks-a$252Fapplication/log-events/ip-10-0-5-104.ap-southeast-1.compute.internal-application.var.log.containers.risingwave-compute-0_rwc-3-longevity-20221103-204020_compute-8cce0cb2e237288c9df034d1609211d57f43ec18b08e45da1890deab316a9dfe.log).

cc @wenym1 Could you please take a look? ❤️

**Comment by hzxa21** [comment]
#6072 this PR may fix the issue. We can priortize it if it is ready.

**Comment by wenym1** [comment]
Uploader is merged in #6072. The code causing the panic will not be reached in the default state store setting (hummock V2). We may close this issue now.
